examples/platforms/utils/soft_source_match_table.c

```diff
@@ -0,0 +1,266 @@
+/*
+ *  Copyright (c) 2019, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file implements a software Source Match table, for radios that don't have
+ *   such hardware acceleration. It supports only the single-instance build of
+ *   OpenThread.
+ *
+ */
+
+#include <openthread-core-config.h>
+#include <openthread/platform/radio.h>
+
+#include <stdlib.h>
+
+#include "common/logging.hpp"
+#include "utils/code_utils.h"
+
+#if defined(RADIO_CONFIG_SRC_MATCH_SHORT_ENTRY_NUM) || defined(RADIO_CONFIG_SRC_MATCH_EXT_ENTRY_NUM)
+static uint16_t sPanId = 0;
+
+void utilsSoftSrcMatchSetPanId(uint16_t aPanId)
+{
+    sPanId = aPanId;
+}
+#endif // defined(RADIO_CONFIG_SRC_MATCH_SHORT_ENTRY_NUM) || defined(RADIO_CONFIG_SRC_MATCH_EXT_ENTRY_NUM)
+
+#ifdef RADIO_CONFIG_SRC_MATCH_SHORT_ENTRY_NUM
```

Comment:
It is better to change line 54 to `#if RADIO_CONFIG_SRC_MATCH_SHORT_ENTRY_NUM`. Otherwise, when user defines macro like `#define RADIO_CONFIG_SRC_MATCH_SHORT_ENTRY_NUM 0`, the code between line 55 and 153 will still be compiled.

Reply:
Changed.